### PR TITLE
Adding barrel lenght to gun parameters

### PR DESCRIPTION
**Pull request by CatWithFourEyes**
#### Summary 

Balance "Replacing gun damage modifiers with barrel lenght, to use it as source for damage calculation. "

#### Purpose of change

Adding new gun parameter, barrel lenght, to parameters, inteded to work with https://github.com/CleverRaven/Cataclysm-DDA/pull/64934 PR for using barrel lenght as damage modifier for guns

#### Describe the solution

Replacing old "ranged_damage": { "damage_type": "bullet", "amount": X }, with "barrel_length": "X mm", 

#### Testing

Can't test it since it's addition to another PR

#### Additional context

List of weapon with added BL and links to sourses of information in pinned file
[Proofs.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/11522428/Proofs.txt)

### CODE REVIEW

File: data/json/items/gun/9mm.json [hunk 1]
```diff
@@ -751,18 +756,17 @@
     "type": "GUN",
     "reload_noise_volume": 10,
     "name": { "str": "STEN Mk II", "str_pl": "STENs Mk II" },
-    "description": "The standard issue 9mm submachine gun of the British military throughout most of the Second World War.  The STEN is aesthetically far from pretty.  Featuring an iconic side mounted magazine well and a design optimized for mass production using unskilled labor, the STEN was developed to make up for drastic supply shortages during the early days of the war.  Likely entering the US prior to the 1960s, either as a war time holdover or a museum purchase; you suppose that if you were to try and defend your nation, whatever the cost may be, and fight the undead ‘on the beaches, fight on the landing grounds, fight in the fields and in the streets, fight in the hills; and never surrender’, then it might be the time for the STEN to come back into frontline service.",
```
Comment (irwiss) [inline data/json/items/gun/9mm.json:754]: intended?

File: data/json/items/gun/nail.json [hunk 2]
```diff
@@ -24,8 +24,7 @@
     "reload": 50,
     "valid_mod_locations": [ [ "grip mount", 1 ], [ "rail mount", 1 ], [ "sights mount", 1 ], [ "stock mount", 1 ], [ "underbarrel mount", 1 ] ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "nail": 20 } } ],
-    "melee_damage": { "bash": 6 },
-    "flags": [ "WONT_TRAIN_MARKSMANSHIP" ]
```
Comment (irwiss) [inline data/json/items/gun/nail.json:28]: this reverts a fix

File: data/json/items/gun/45.json [hunk 3]
```diff
@@ -367,10 +371,10 @@
     "color": "brown",
     "ammo": [ "45" ],
     "skill": "smg",
-    "ranged_damage": { "damage_type": "bullet", "amount": 2 },
     "dispersion": 360,
     "durability": 7,
     "min_cycle_recoil": 540,
+    "default_mods": [ "muzzle_brake_mod" ],
```
Comment (irwiss) [inline data/json/items/gun/45.json:377]: The tommy gun photo on wikipedia doesn't look like it has a brake, is this sourced from elsewhere?

File: data/json/items/gun/shot.json [hunk 4]
```diff
@@ -382,6 +388,7 @@
         "name": { "str": "Remington 1100 competition" },
         "description": "This semi-automatic shotgun features a self compensating gas system that will feed a wide array of shells while also reducing recoil.  Introduced in 1963, it is favored by law enforcement, hunters and competition shooters, and has been the best-selling autoloading shotgun in U.S. history.  This is the nickel finished, teflon coated competition model, with a full length magazine tube and 30 inch barrel.",
         "ascii_picture": "remington1100"
+		"barrel_length": "762 mm",
```
Comment (irwiss) [inline data/json/items/gun/shot.json:391]: can variants define barrel_length? glancing at c++ side doesn't look like it

File: data/json/items/gun/45.json (hunk 3 above)
Comment (CatWithFourEyes) [reply data/json/items/gun/45.json:377]: ingame M1928A1 doesn't have it, so i thinking about "chekup" on weapon mods later. It's not a single case, many guns have integrated muzzle brakes, some have "longest side" of folded gun, while having folded stock.

File: data/json/items/gun/9mm.json (hunk 1 above)
Comment (CatWithFourEyes) [reply data/json/items/gun/9mm.json:754]: not sure. Looks like typo hunting

File: data/json/items/gun/nail.json (hunk 2 above)
Comment (CatWithFourEyes) [reply data/json/items/gun/nail.json:28]: I see, was patched while i doing barrel lenght at night

File: data/json/items/gun/shot.json (hunk 4 above)
Comment (Maleclypse) [reply data/json/items/gun/shot.json:391]: They cannot

**Comment by DoctorBoomstick** [comment]
You’re lacking a summery entry in your PR.
At the top of your PR description you should include something like:
#### Summery
Content “a summery of your changes goes here in the quotes”
You also want to lint your changes.
http://dev.narc.ro/cataclysm/format.html
copy and paste the contents of every file you’ve touched one by one into the editor in the provided link, hit the lint button, copy the results from the editor (even if it looks like it hasn’t changed anything) and paste it back into the original file. Repeat this for every file in which you’ve made changes. The linter makes sure your changes are formatted correctly.

File: data/json/items/gun/nail.json (hunk 2 above)
Comment (CatWithFourEyes) [reply data/json/items/gun/nail.json:28]: Removing a bash damage also was intended?

Comment (irwiss) [reply data/json/items/gun/nail.json:28]: melee damage is still there even in your PR, but shows as a changed line due to comma - the ancient json spec used doesn't allow trailing commas, so you'll see changes like that when you change last line of json object

**Comment by TheShadowFerret** [comment]
Will defining barrel length on bullpups let me cut inside where the hacksaw can reach 🤔

File: data/json/items/gun/shot.json (hunk 4 above)
Comment (CatWithFourEyes) [reply data/json/items/gun/shot.json]: So, what is better, delete this lines, or leave for the time?

Comment (irwiss) [reply data/json/items/gun/shot.json]: It likely won't pass tests if you leave random bits of unused json, unless the specific json loader allows for it (don't count on it, as most don't)

**Comment by DoctorBoomstick** [comment]
@TheShadowFerret , If there isn’t a barrel_volume entry listed, then it shouldn’t. bullpups should probably not have this because the notion of sawing down a bullpup is funny. Tbh, a lot of gas-operated rifles shouldn’t have this either.

**Comment by DoctorBoomstick** [comment]
Your summery entry is formatted incorrectly. It should be enclosed in quotes, and prefixed with the words, “Content, Balance, or Bugfix.”
Prefixing it in this way lets the changelog creator know which category to slot your pr in when it gets merged. So, if you wanted it to be listed under content, it should look something like,:
Content “Replacing gun damage modifiers with barrel lenght, to use it as source for damage calculation.”

**Comment by GuardianDll** [comment]
you accidentally remove two big files it seems

**Comment by CatWithFourEyes** [comment]
> you accidentally remove two big files it seems

Don't even know why. 

Turning into a draft, since now it's just a sourse of information.

File: data/json/items/gun/22.json [hunk 5]
```diff
@@ -167,6 +167,7 @@
     "weight": "1953 g",
     "volume": "1610 ml",
     "longest_side": "1000 mm",
+    "barrel_length": "660 mm",
```
Comment (TheShadowFerret) [inline data/json/items/gun/22.json:170]: These are made from 60cm pipes

**Comment by GuardianDll** [comment]
is there any reason preventing you from releasing it?

**Comment by Maleclypse** [comment]
> is there any reason preventing you from releasing it?

It looks like it has several json errors.

**Comment by CatWithFourEyes** [comment]
> is there any reason preventing you from releasing it?

From the start of it, it was more of "groundwork" for barrel lenght system as DMG modifier. And since some guns were redone from the ground, like AR-15 system, i would prefer to leave it as information source

**Comment by Maleclypse** [comment]
> > is there any reason preventing you from releasing it?
> 
> From the start of it, it was more of "groundwork" for barrel lenght system as DMG modifier. And since some guns were redone from the ground, like AR-15 system, i would prefer to leave it as information source

So you are saying you want it closed because it’s never going to be in a merge ready state? I’m trying to understand what the above means.

**Comment by CatWithFourEyes** [comment]
> it’s never going to be in a merge ready state

Exactly. It's more of data source.

**Pull request closed without merging**